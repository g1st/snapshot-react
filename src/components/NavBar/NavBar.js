import React from 'react';

const NavBar = () => {
  return <div>im NavBar component</div>;
};

export default NavBar;
