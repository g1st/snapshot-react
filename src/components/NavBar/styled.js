import styled from 'styled-components';
import { lighten } from 'polished';
import { NavHashLink as NavLink } from 'react-router-hash-link';
// import { NavLink } from 'react-router-dom';

import { colorMainDark, colorText } from '../../styles/variables';

export const Wrapper = styled.div`
  min-height: 58px;
  width: 100%;
  background-color: #fff;
  position: fixed;
  top: 0;
  left: 0;
  z-index: 4;
  overflow: hidden;
`;
Wrapper.displayName = 'Wrapper';

export const NavigationBar = styled.nav`
  height: ${props => (props.menuExpanded ? '260px' : 0)};
  display: grid;
  max-width: 1400px;
  width: 100%;
  grid-template-areas: 'logo' 'navigation';
  transition: height 0.3s ease-out;
  margin: 6px auto 0 auto;
  grid-template-rows: auto 260px;
  grid-row-gap: 10px;
  @media (min-width: 768px) {
    grid-template-areas: 'logo navigation';
    height: 0;
  }
`;
NavigationBar.displayName = 'NavigationBar';

export const Logo = styled.div`
  font-size: 2.1rem;
  font-weight: bold;
  color: ${colorMainDark};
  margin-left: 20px;
  grid-area: logo;
  align-self: top;
  @media (min-width: 768px) {
    align-self: center;
  }
`;
Logo.displayName = 'Logo';

export const NavBarItems = styled.ul`
  list-style: none;
  display: block;
  grid-area: navigation;
  justify-self: start;
  width: 100%;
  border-top: 1px solid ${colorText};
  & > :first-child {
    margin-top: 12px;
  }

  @media (min-width: 768px) {
    align-self: center;
    justify-self: end;
    display: flex;
    width: auto;
    border: none;
    & > :first-child {
      margin-top: 0;
      margin-left: 0;
    }
  }
`;
NavBarItems.displayName = 'NavBarItems';

export const NavLinkItem = styled(NavLink)`
  display: block;
  position: relative;
  color: ${lighten(0.4, `${colorMainDark}`)};
  padding: 5px 0 5px 20px;
  text-transform: uppercase;
  font-size: 0.9rem;
  width: 100%;
  line-height: 36px;
  text-decoration: none;

  :hover {
    cursor: pointer;
    color: ${colorMainDark};
  }

  @media (min-width: 768px) {
    display: inline-block;
    font-size: 1rem;
    margin: 0 30px;
    padding-left: 0;

    :hover {
      color: #9aaebf;
    }

    ::before {
      content: '';
      position: absolute;
      height: 2px;
      width: 100%;
      background-color: rgba(0, 0, 0, 0.5);
      bottom: 5px;
      left: 0;
      transform: scaleX(0);
      transition: all 0.4s ease;
      visibility: hidden;
      ${props =>
        props.active === 'true'
          ? `{
      transform: scaleX(1);
      visibility: visible;
    }`
          : null};
    }
  }
`;
NavLinkItem.displayName = 'NavLinkItem';

export const Menu = styled.div`
  display: block;
  position: absolute;
  top: 18px;
  right: 0;
  user-select: none;
  margin-right: 20px;
  cursor: pointer;

  @media (min-width: 768px) {
    display: none;
  }
`;
Menu.displayName = 'Menu';
