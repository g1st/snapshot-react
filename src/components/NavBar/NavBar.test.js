import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import NavBar from '.';

describe('NavBar Component', () => {
  it('matches snapshot', () => {
    const tree = shallow(<NavBar />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
