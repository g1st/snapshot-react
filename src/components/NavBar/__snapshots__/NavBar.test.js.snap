// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Component matches snapshot 1`] = `
<Wrapper>
  <Menu
    onClick={[Function]}
  >
    â˜°
  </Menu>
  <NavigationBar
    menuExpanded={false}
  >
    <Logo>
      Snapshot
    </Logo>
    <NavBarItems
      menuExpanded={false}
    >
      <NavLinkItem
        active="true"
        exact={true}
        onClick={[Function]}
        smooth={true}
        to="/#home"
      >
        Home
      </NavLinkItem>
      <NavLinkItem
        active={null}
        onClick={[Function]}
        scroll={[Function]}
        to="/#about"
      >
        About
      </NavLinkItem>
      <NavLinkItem
        active={null}
        onClick={[Function]}
        smooth={true}
        to="/#gallery"
      >
        Gallery
      </NavLinkItem>
      <NavLinkItem
        active={null}
        onClick={[Function]}
        smooth={true}
        to="/#contact"
      >
        Contact
      </NavLinkItem>
    </NavBarItems>
  </NavigationBar>
</Wrapper>
`;
